src/core_plugins/tile_map/public/base_maps_visualization.js

```diff
@@ -99,75 +100,62 @@ export function BaseMapsVisualizationProvider(serviceSettings) {
       this._kibanaMap.on('baseLayer:loading', () => {
         this._baseLayerDirty = true;
       });
-
-      const mapparams = this._getMapsParams();
-      await this._updateBaseLayer(mapparams);
+      await this._updateBaseLayer();
     }
 
 
-    async _updateBaseLayer(mapParams) {
+    async _updateBaseLayer() {
 
-      try {
-        this._tmsService = await serviceSettings.getTMSService();
-        this._tmsError = null;
-      } catch (e) {
-        this._tmsService = null;
-        this._tmsError = e;
-        this._notify.warning(e.message);
+      const mapParams = this._getMapsParams();
+      if (!mapParams.wms.baseLayersAreLoaded && !mapParams.wms.selectedTmsLayer) {
+        return;
       }
-      const { minZoom, maxZoom } = this._getMinMaxZoom();
-      if (mapParams.wms.enabled) {
-        // Switch to WMS
-        if (maxZoom > this._kibanaMap.getMaxZoomLevel()) {
-          //need to recreate the map with less restrictive zoom
-          this._kibanaMap.removeLayer(this._geohashLayer);
-          this._geohashLayer = null;
-          this._kibanaMap.setMinZoom(minZoom);
-          this._kibanaMap.setMaxZoom(maxZoom);
-        }
 
-        this._kibanaMap.setBaseLayer({
-          baseLayerType: 'wms',
-          options: {
-            minZoom: minZoom,
-            maxZoom: maxZoom,
-            url: mapParams.wms.url,
-            ...mapParams.wms.options
-          }
-        });
-      } else {
-
-        // switch to tms
-        if (maxZoom < this._kibanaMap.getMaxZoomLevel()) {
-          //need to recreate the map with more restrictive zoom level
-          this._kibanaMap.removeLayer(this._geohashLayer);
-          this._geohashLayer = null;
-          this._kibanaMap.setMinZoom(minZoom);
-          this._kibanaMap.setMaxZoom(maxZoom);
-          if (this._kibanaMap.getZoomLevel() > maxZoom) {
-            this._kibanaMap.setZoomLevel(maxZoom);
+      try {
+
+        if (mapParams.wms.enabled) {
+          const minZoom = 0;
```

Comment:
should define on top and use just once, or even better use defaults set on vis.params ?